File: agricore_spring_boot/src/main/java/agricore/projet/dto/data/TransformationDataDTO.java
package agricore.projet.dto.data;

import agricore.projet.model.ressource.Transformation;
import agricore.projet.model.zone.NomZone;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.List;

public record TransformationDataDTO(
        Transformation transformation,
        NomZone requiredZone,
        boolean zoneExist,
        int maxTransfoPossible,
        List<TransformationPartDTO> ingredients,
        List<TransformationPartDTO> produits
) {}

File: agricore-angular/src/app/dto/ressource/transformation-data-dto.ts
import {TransformationPartDto} from './transformation-part-dto';

export interface TransformationDataDto {
  transformation: string;
  requiredZone: string;
  zoneExist: boolean;
  maxTransfoPossible: number;
  ingredients: TransformationPartDto[];
  produits: TransformationPartDto[];
}